### PR TITLE
Issue#186 function names refactoring

### PR DESCRIPTION
**Pull request by kbaird**

### CODE REVIEW

**Comment by kbaird** [comment]
@elbrujohalcon I think this is ready for comments.

**Comment by elbrujohalcon** [comment]
@kbaird can you add your rule to the [elvis.config](https://github.com/inaka/elvis/blob/master/config/elvis.config) file, please?

**Comment by elbrujohalcon** [comment]
and if you haven't already, please add your rule to [the wiki](https://github.com/inaka/elvis/wiki/Rules) as well. Thanks :)

File: config/elvis.config [hunk 1]
```diff
@@ -26,6 +26,11 @@
                     #{regex => "^([a-z][a-z0-9]*_?)*(_SUITE)?$",
                       ignore => []}
                   },
+                  {
+                    elvis_style,
+                    function_naming_convention,
+                    #{regex => "^([a-z]*_?)*(_SUITE)?$"}
```
Comment (elbrujohalcon) [inline config/elvis.config]: let's change that regex to something that says _"lowercase ascii charts, underscores and digits"_, which is usually the default value for this rule, please

Comment (kbaird) [inline config/elvis.config]: I intentionally removed digits based on "Function names must use only lowercase characters" at https://github.com/inaka/erlang_guidelines#function-names. I also have a rule that forbids digits in function names.

I just want to confirm that you'd like that rule removed, and the regex to match the same one used for module names, which allows digits?

If so, can we update the quotation that I misinterpreted to say "only lowercase characters or digits" ?

Comment (elbrujohalcon) [inline config/elvis.config]: We can update the quotation. Would you mind sending a PR with that change?

When I think of functions with numbers, I'm thinking in stuff like `base64_encode` or `rfc###_…` or `iso6801_…`, etc.

Comment (kbaird) [inline config/elvis.config]: Sure. pushing the change now, and will do that PR for the guidelines after that. Thanks.

Comment (kbaird) [inline config/elvis.config]: [Guidelines PR](https://github.com/inaka/erlang_guidelines/pull/58)

File: config/elvis.config [hunk 2]
```diff
@@ -26,6 +26,11 @@
                     #{regex => "^([a-z][a-z0-9]*_?)*(_SUITE)?$",
                       ignore => []}
                   },
+                  {
+                    elvis_style,
+                    function_naming_convention,
+                    #{regex => "^([a-z][a-z0-9]*_?)*(_SUITE)?$"}
```
Comment (elbrujohalcon) [inline config/elvis.config]: you still need to remove `_SUITE` from here :trollface:

Comment (kbaird) [inline config/elvis.config]: Why is the `(_SUITE)` present for the `module_naming_convention`? I was trying to follow the pattern established from module names by `(_SUITE)` being present [in elvis.config](https://github.com/kbaird/elvis/blob/Issue%23186-FunctionNames-Refactoring/config/elvis.config#L26) but absent from [style_SUITE.erl](https://github.com/kbaird/elvis/blob/Issue%23186-FunctionNames-Refactoring/test/style_SUITE.erl#L290)

If `(_SUITE)` is removed from [the function use](https://github.com/kbaird/elvis/blob/Issue%23186-FunctionNames-Refactoring/config/elvis.config#L32), should it also be removed from [the module use](https://github.com/kbaird/elvis/blob/Issue%23186-FunctionNames-Refactoring/config/elvis.config#L26)?

Comment (elbrujohalcon) [inline config/elvis.config]: because common-test modules *must* be called `something_SUITE`

Comment (kbaird) [inline config/elvis.config]: OK, cool. Thanks. I was genuinely curious. Fix coming soon.

**Pull request merged**